iothub/service/src/Messaging/MessagingClient.cs

```diff
@@ -0,0 +1,379 @@
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Closing MessagingClient", nameof(CloseAsync));
+
+            try
+            {
+                await _faultTolerantSendingLink.CloseAsync().ConfigureAwait(false);
+                await _connection.CloseAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(CloseAsync)} threw an exception: {ex}", nameof(CloseAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Closing MessagingClient", nameof(CloseAsync));
+            }
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Disposing MessagingClient", nameof(Dispose));
+
+            _faultTolerantSendingLink.Dispose();
+            _connection.Dispose();
+
+            if (Logging.IsEnabled)
+                Logging.Exit(this, $"Disposing MessagingClient", nameof(Dispose));
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Send a cloud-to-device message to the specified device.
+        /// </summary>
+        /// <param name="deviceId">The device identifier for the target device.</param>
+        /// <param name="message">The cloud-to-device message.</param>
+        /// <param name="timeout">The operation timeout, which defaults to 1 minute if unspecified.</param>
+        /// <exception cref="ArgumentNullException">Thrown when the provided <paramref name="deviceId"/> or <paramref name="message"/> is null.</exception>
+        /// <exception cref="ArgumentException">Thrown if the provided <paramref name="deviceId"/> is empty or whitespace.</exception>
+        /// <exception cref="IotHubException">
+        /// Thrown if IoT hub responded to the request with a non-successful status code. For example, if the provided
+        /// request was throttled, <see cref="IotHubThrottledException"/> is thrown. For a complete list of possible
+        /// error cases, see <see cref="Common.Exceptions"/>.
+        /// </exception>
+        public virtual async Task SendAsync(string deviceId, Message message, TimeSpan? timeout = null)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Sending message with Id [{message?.MessageId}] for device {deviceId}", nameof(SendAsync));
+
+            Argument.RequireNotNullOrEmpty(deviceId, nameof(deviceId));
+            Argument.RequireNotNull(message, nameof(message));
+
+            if (_clientOptions?.SdkAssignsMessageId == SdkAssignsMessageId.WhenUnset && message.MessageId == null)
+            {
+                message.MessageId = Guid.NewGuid().ToString();
+            }
+
+            if (message.IsBodyCalled)
+            {
+                message.ResetBody();
```

Comment:
This looks like it is only necessary if we support users passing in Streams as the payload. If we just take byte[], then there is no need for this.

Reply:
this change is added in the PR for removing Disposable dependencies from Message API. removing it from this PR as well.